src/cow-react/modules/account/containers/Transaction/ActivityDetails.tsx

```diff
@@ -164,6 +168,21 @@ export function ActivityDetails(props: {
   const showProgressBar = (activityState === 'open' || activityState === 'filled') && order?.class !== 'limit'
   const showCancellationModal = activityDerivedState.order ? getShowCancellationModal(activityDerivedState.order) : null
 
+  const { surplusAmount, surplusToken } = useMemo(() => {
+    const output: { surplusToken?: Token; surplusAmount?: CurrencyAmount<Token> } = {}
+
+    if (order) {
+      const summaryData = getExecutedSummaryData(order)
+      output.surplusAmount = summaryData.surplusAmount
+      output.surplusToken = summaryData.surplusToken
+    }
+
+    return output
+  }, [JSON.stringify(order)])
+
+  const fiatValue = useCoingeckoUsdValue(surplusAmount)
```

Comment:
In the orders modal the current spot price from coingecko is fine, it's happening right now.

Showing this here might be more tricky.
0.1 WETH today is not the same 6 months ago.

Take a look for example on etherscan

It shows the current fee and on click shows the fee at execution time

![Screenshot 2023-04-20 at 16 19 04](https://user-images.githubusercontent.com/43217/233411822-683fd6e4-daa0-4e65-96e2-67ca61b3904f.png)

![Screenshot 2023-04-20 at 16 19 12](https://user-images.githubusercontent.com/43217/233411837-df151cde-1efe-48ef-bea5-e1421aac3148.png)

Can we somehow show the surplus at execution time?

Reply:
@alfetopito Hmm I'm not sure I understand. 
We are showing the same surplus that we show in order receipt and the explorer.
And then for USD value we just calculate it from that surplus value.

Reply:
> We are showing the same surplus that we show in order receipt and the explorer.
Yes, that's correct and the surplus amount will never change.

What will change over time is how much 1ETH is worth today (and whatever else you got the surplus on)

Thus, looking at the usd value of 0.1ETH now vs 6 months ago you'll get different results

Reply:
Oh I see, yea I've raised that question on daily before we decided to implement this.
But the only way to resolve this would be to store the USD value of surplus at the execution time the way I see it but I don't think we want to do that now. So this is the next best thing I guess.